### DISCUSSION TITLE
NuxtJS, /api route always return 404

### DISCUSSION THREAD

**Discussion by arhakim** [vercel/vercel#4030] [General] [answered]
I run the `now dev` command, its compiled successfully but when I tried to call the `/api` route e.g `http://localhost:3000/api/auth/login` it's always return 404 page not found.

It's work successfully when I run `npm run dev` thought

Btw, I am using NuxtJS and ExpressJS

Here is my `now.json`
```
{
  "version": 2,
  "functions": {
	"api/index.js": {
		"memory": 3008,
		"maxDuration": 60
	}
  },
  "routes":[
	{"src": "/api", "dest": "/api/index?input=$1"}
  ]
}
```

**Comment by arhakim** [comment]
Still got 404 error even I change the routes to:
```
...
"routes":[
	{"src": "/api", "dest": "/api/index.js"}
  ]
...
```

**Comment by styfle** [comment, marked as answer]
Hi @arhakim

You don't need `{"src": "/api", "dest": "/api/index.js"}` because that is how routing already works. If an index file is found in the directory then it will be served. Visiting these paths will all serve the same function:

- `/api`
- `/api/index`
- `/api/index.js`

I would recommending converting your Express app to proper Serverless Functions by following this Guide: https://zeit.co/guides/migrate-to-zeit-now#serverful-vs.-serverless-apis

**Comment by arhakim** [comment]
> I would recommending converting your Express app to proper Serverless Functions by following this Guide: https://zeit.co/guides/migrate-to-zeit-now#serverful-vs.-serverless-apis

This fix the issue.

Thank you!
